### PR TITLE
[FEAT] [DOCS] Refactor add and update records methods in favor of log method with optional fields

### LINKED ISSUE #226: [FEATURE] Allow updating record attributes via a Record class definition
Currently, the field attribute of a record is not optional, so attributes like vectors and metadata cannot be updated alone. The user has to use a dictionary definition.

This should support flows like this:

```python
dataset.records.update(Record(vectors=[rg.Vector(...)]))
```

### LINKED ISSUE #228: [DOCS] Improve `dataset.record.add` documentation so that it clarifies when to use `Resource` or dict
Currently the documentation is centered on adding and updating records as dictionaries and does not explain when it is better to use a `Resource` class.

This is misleading to users because it encourages them down a route that lacks type hinting and doc strings.

### LINKED ISSUE #229: [REFACTOR] `add`/`update` >> `log`
The `update` method of `DatasetRecords` adds new records which makes the `add` method redundant. We could reduce documentation and code by using a single method that we explain fully.

### PR DESCRIPTION
**Pull request by burtenshaw**
This feature unifies the previous add/ update method under a dingle method `add`.

Here are the main changes:
- renames update to add and deprecates the old update
- stops the support for single instances of a record or dict
- updates the api reference for updating with `add` method
- updates the api reference to start from the `rg.Record` approach and uses tabs for other options
- updates the how to on records as above and adds more explanation
- includes an example of adding records from a HF dataset using the integration

### CODE REVIEW

**Comment by burtenshaw** [comment]
@sdiazlor @davidberenstein1957 There are changes to the how-to's and API references. I merge #224 if that's easier to review.

**Comment by burtenshaw** [comment]
@nataliaElv It would be great if you could take a look at the docs changes and see if they reflect our recent conversations with users.

File: src/argilla_sdk/records/_resource.py [hunk 1]
```diff
@@ -76,6 +76,12 @@ def __init__(
             _server_id: An id for the record. (Read-only and set by the server)
             _dataset: The dataset object to which the record belongs.
         """
+
+        if fields is None and metadata is None and vectors is None and responses is None and suggestions is None:
```
Comment (davidberenstein1957) [inline src/argilla_sdk/records/_resource.py:80]:
```suggestion
    if all(x is None for x in [fields, metadata, vectors, responses, suggestions]):
        raise ValueError("At least one of fields, metadata, vectors, responses, or suggestions must be provided.")
```

File: src/argilla_sdk/records/_dataset_records.py [hunk 2]
```diff
@@ -373,8 +325,6 @@ def _ingest_records(
         mapping: Optional[Dict[str, str]] = None,
         user_id: Optional[UUID] = None,
     ) -> List[RecordModel]:
-        if isinstance(records, (Record, dict)):
```
Comment (davidberenstein1957) [inline src/argilla_sdk/records/_dataset_records.py:376]: the ValueError message below has not been updated

File: src/argilla_sdk/records/_resource.py [hunk 3]
```diff
@@ -76,6 +76,12 @@ def __init__(
...
+            raise ValueError("At least one of fields, metadata, vectors, responses, or suggestions must be provided.")
+        if fields is None and id is None:
```
Comment (davidberenstein1957) [inline src/argilla_sdk/records/_resource.py:82]: perhaps we can also use the same logic as above suggestion to unify?

**Review by davidberenstein1957 — COMMENTED** [review]
Hi @burtenshaw,

Looks good. Not sure if there is a better way to explain the footer behavior within the code refs added through (1) and (2)

File: docs/how_to_guides/record.md [hunk 4]
```diff
@@ -444,7 +418,7 @@ updated_data = [
     }
     for sample in data
 ]
-dataset.records.update(records=updated_data)
+dataset.records.add(records=updated_data)
 ```
 !!! note "Update the metadata"
     To update the metadata of a record, you can iterate over the records and update the metadata dictionary by key or using `metadata.update`. After that, you should update the records in the dataset.
```
Comment (nataliaElv) [inline docs/how_to_guides/record.md:424]: `metadata.update` in this paragraph should be changed to `metadata.add` or `metadata.log` (depending on the final name). Same for the paragraph right below the "Update records" heading.

File: src/argilla_sdk/records/_dataset_records.py [hunk 5]
```diff
@@ -345,8 +298,7 @@ def from_json(self, path: Union[Path, str]) -> "DatasetRecords":
 
         """
         records = JsonIO._records_from_json(path=path)
-        return self.update(records=records)
-        return self
+        return self.add(records=records)
```
Comment (frascuchon) [inline src/argilla_sdk/records/_dataset_records.py:301]: method signature should be changed to `) ->  List[Record]:`

**Review by frascuchon — COMMENTED** [review]
My only general comment is about the `add` naming. +1 to `records.log`. But, in that case, we should change the `LoggingMixin.log` method name.

**Comment by burtenshaw** [comment]
> My only general comment is about the `add` naming. +1 to `records.log`. But, in that case, we should change the `LoggingMixin.log` method name.

`LoggingMixin.log` >> `_log_message`

**Review by davidberenstein1957 — COMMENTED** [review]
@burtenshaw perhaps it is nice to add the code comments as tooltips? https://squidfunk.github.io/mkdocs-material/reference/tooltips/, or does (1) actually translate to this, then is perfect like this? https://squidfunk.github.io/mkdocs-material/reference/annotations/

File: src/argilla_sdk/_helpers/_log.py [hunk 6]
```diff
@@ -24,11 +24,20 @@
 }
 
 
-def log(message: str, level: str = "info") -> None:
+def log_message(message: str, level: str = "info") -> None:
```
Comment (davidberenstein1957) [inline src/argilla_sdk/_helpers/_log.py:27]: perhaps we can also give this file another name? _logger/_logging?

File: src/argilla_sdk/records/_resource.py (hunk 1 above)
Comment (davidberenstein1957) [reply src/argilla_sdk/records/_resource.py:79]: @burtenshaw did you miss this? or did not want to do this?

Comment (burtenshaw) [reply src/argilla_sdk/records/_resource.py:79]: @davidberenstein1957 Sorry not to mention. I'm not into the creation of a generator and data structure for a logical expression.

**Review by frascuchon — APPROVED** [review]
Nice! Just to don't forget to review the PR title before merge it.

**Pull request merged**